### ISSUE TITLE
[xtext-eclipse] overload DefaultDescriptionLabelProvider#image() for EClass

### ISSUE THREAD

**Issue by knutwannheden** [eclipse/xtext-eclipse#85]
DefaultDescriptionLabelProvider is the base class for the stub generated by the default LabelProviderFragment. It implements image(IEObjectDescription) like this:

```
public Object image(IEObjectDescription element) {
    return image(element.getEClass());
}
```

Unfortunately there is no overloaded image(EClass) so it goes to image(Object).

I think it would make sense with an overloaded image(EClass) and even have a default implementation for it, which can be overridden:

```
@Override
public Object image(final EClass element) {
    return convertToImageDescriptor("full/obj16/" + element.getName() + ".gif");
}
```

Alternatively image(IEObjectDescription) could be changed to:

```
public Object image(IEObjectDescription element) {
    return doGetImage(element.getEClass());
}
```

That way the user could still declare image(EClass) in the generated stub and it would be called.

**Comment by cdietrich** [comment]
i prefer the doGetImage variant

**Comment by cdietrich** [comment]
fixed in 2.13
